core-api/src/main/java/com/optimizely/ab/config/audience/match/MatchType.java

```diff
@@ -28,7 +29,7 @@
     private String matchType;
     private Match matcher;
 
-    public static MatchType getMatchType(String matchType, Object conditionValue) {
+    public static MatchType getMatchType(String matchType, Object conditionValue, UserAttribute userAttribute) {
```

Comment:
:-1: to adding an argument to the signature just for logging

Reply:
@loganlinn Do you have any suggestion to deal it, where we need to log with userAttributes. I am thinking to add some enum types for NullMatch, when NullMatch is returned, we'll also set NullMatchReasonType in the object. So In two or many more different cases we will check it using ReasonType under NullMatch object.